DevelopGitHub/GitHubExtension/GitHubExtention.Preferences.WebApi/Controllers/UserPreferencesController.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Data.Entity;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+using GitHubExtension.Infrastructure.Constants;
+using GitHubExtension.Preferences.DAL.Model;
+using GitHubExtention.Preferences.WebApi.Constants;
+using GitHubExtention.Preferences.WebApi.Extensions;
+using Microsoft.AspNet.Identity;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace GitHubExtention.Preferences.WebApi.Controllers
+{
+    [RoutePrefix(RouteConstants.ApiUser)]
+    public class UserPreferencesController : ApiController
+    {
+        private readonly PreferencesContext _context;
+
+        private readonly AzureBlobStorageMultipartProvider _multipartStreamProvider;
+
+        public UserPreferencesController(PreferencesContext context, AzureBlobStorageMultipartProvider provider)
+        {
+            _context = context;
+            _multipartStreamProvider = provider;
+        }
+
+        [HttpPost]
+        [AllowAnonymous]
+        [Route(RouteConstants.Avatar)]
+        public async Task<IHttpActionResult> Post()
+        {
+            if (!Request.Content.IsMimeMultipartContent(AvatarConstants.FormData))
+            {
+                throw new HttpResponseException(HttpStatusCode.UnsupportedMediaType);
+            }
+
+            User user = _context.GetUser(User.Identity.GetUserId());
+            if (user == null)
+            {
+                return NotFound(); 
+            }
+
+            await Request.Content.ReadAsMultipartAsync<AzureBlobStorageMultipartProvider>(_multipartStreamProvider);
```

Comment:
Please don't mix persistence layer with Azure complexity...It should be separarted btw DbContext also support async operations.